src/analyses/apron/apronAnalysis.apron.ml

```diff
@@ -146,7 +146,7 @@ struct
     if !GU.global_initialization && e = MyCFG.unknown_exp then
       st (* ignore extern inits because there's no body before assign, so the apron env is empty... *)
     else (
-      if M.tracing then M.traceli "apron" "assign %a = %a\n" d_lval lv d_exp e;
+      if M.tracing then M.traceli "apron" "assign %a = %a\n" RenameMapping.d_lval lv d_exp e;
```

Comment:
I was also thinking that, since the constraint system is generated from the new CFG, it is not sufficient to rename a `varinfo` from the old Cil file. But this could actually still work, because of the substitution of the locals, formals, statements and so on, that is done for unchanged functions (in the code snippet I linked above). One only needs to do something similar for the partially changed functions (CFG comparison) also. We should check anyway, why this is not the case yet.

Reply:
> And I see no point why locations should be handled any different then? Could we not also just change the location of the old data structures to contain the new one?

The fields of `Cil.location` are not mutable right now at least: in `Cil.instr` and `Cil.stmtkind` they are just immutable components of constructor argument tuples. If they were, then it also should be possible, but that requires significant changes to CIL itself.
And the output of locations happens in much fewer places in Goblint code, so the location remapping isn't really noticeable in most code.

Reply:
I have been playing around with simply renaming the varinfos in the old AST as you @sim642 suggested. [Here is my code that performs the renaming](https://github.com/TimOrtel/analyzer/blob/dev-local-var-rename-without-rename-mapping/src/incremental/updateCil.ml#L46). This works in most cases, e.g. in g2html the graph displays the correctly names. However, if you click on the details of each node in g2html, the renamings are not present there, and the old/original names are displayed there. The code responsible for printing the varinfo names in this detail view is [basetype.ml#Variables#show](https://github.com/goblint/analyzer/blob/master/src/cdomains/basetype.ml#L30). 

Probably I am missing something, but to me it looks like that there are at least two copies of the varinfos, because otherwise my renaming should have affected the vname. I am unsure how I can also rename these instances, because I couldn't quite figure out where these are coming from.

Reply:
> Probably I am missing something, but to me it looks like that there are at least two copies of the varinfos, because otherwise my renaming should have affected the vname.

Yeah, that looks like it! Just a hunch: what happens if you turn off hashconsing? There should be a switch `ana.opt.hashcons` or something similar...

Reply:
Running test-incremental with an additional argument --disable ana.opt.hashcons does not change anything in the g2html output.

Reply:
ok, was worth a try!

Reply:
I think the big reason why it doesn't work yet is #357. Right now `incremental.save` will marshal the old `Cil.file` and the solution to separate files. When `incremental.load` unmarshals them, the `varinfo`s in the loaded AST and the loaded solution are no longer shared, so modifying them in the AST doesn't modify them in the solution (including the local values for nodes in g2html).

Marshaling everything into a single file should keep the physical sharing on unmarshaling and allow this to work fully.

Reply:
>When `incremental.load` unmarshals them, the `varinfos` in the loaded AST and the loaded solution are no longer shared, so modifying them in the AST doesn't modify them in the solution (including the local values for nodes in g2html).

This should be fixed with #748 being merged into `master`.